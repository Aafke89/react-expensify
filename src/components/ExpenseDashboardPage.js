import React from 'react';
import ExpenseList from './ExpenseList';
import ExpensListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary /> 
        <ExpensListFilters /> 
        <ExpenseList /> 
    </div>
)

export default ExpenseDashboardPage;