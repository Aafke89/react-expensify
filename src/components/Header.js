import React from 'react';
import {NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <div>
            <h1>Expensify</h1>
            <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
            <NavLink to="/create" activeClassName="is-active">Create</NavLink>
        </div>
    </header>
);

export default Header; 