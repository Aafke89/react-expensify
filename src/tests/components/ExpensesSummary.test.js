import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('Should return correctly summaryExpenses with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
})

test('Should return correctly summaryExpenses with more expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={23} expensesTotal={2380} />);
    expect(wrapper).toMatchSnapshot();
})
