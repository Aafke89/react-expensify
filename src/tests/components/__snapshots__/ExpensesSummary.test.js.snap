// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return correctly summaryExpenses with more expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $23.80
     
  </h1>
</div>
`;

exports[`Should return correctly summaryExpenses with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
     
  </h1>
</div>
`;
